From 752ea45ee7b8a18f9b952713afe7ae38001def56 Mon Sep 17 00:00:00 2001
From: Wang Hui <Hui.Wang@windriver.com>
Date: Fri, 18 Jan 2013 13:31:40 +0800
Subject: [PATCH 29/50] armadaxp: reset: fix arch_reset according to new kernel api

In the past, the arch_reset() is called by the arm common function
arm_machine_restart(), but in current linux kernel, it should be
set to machine operations.

Signed-off-by: Wang Hui <Hui.Wang@windriver.com>
---
 arch/arm/mach-armadaxp/core.c                |    5 +++++
 arch/arm/mach-armadaxp/include/mach/system.h |    1 -
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-armadaxp/core.c b/arch/arm/mach-armadaxp/core.c
index 5eceb38..c3c67ba 100755
--- a/arch/arm/mach-armadaxp/core.c
+++ b/arch/arm/mach-armadaxp/core.c
@@ -1648,6 +1648,7 @@ MACHINE_START(ARMADA_XP_DB, "Marvell Armada XP Development Board")
 	.init_irq	= axp_init_irq,
 	.timer		= &axp_timer,
 	.init_machine	= axp_db_init,
+	.restart	= arch_reset,
 #ifdef CONFIG_FB_DOVE
 	/* reserve memory for LCD */
 	.fixup		= axp_tag_fixup_mem32,
@@ -1786,6 +1787,7 @@ MACHINE_START(ARMADA_XP_GP, "Marvell Armada XP GP Board")
 	.init_irq	= axp_init_irq,
 	.timer		= &axp_timer,
 	.init_machine	= axp_gp_init,
+	.restart	= arch_reset,
 #ifdef CONFIG_FB_DOVE
 	/* reserve memory for LCD */
 	.fixup		= axp_tag_fixup_mem32,
@@ -1924,6 +1926,7 @@ MACHINE_START(ARMADA_XP_RD_NAS, "Marvell Armada XP RD NAS Board")
 	.init_irq	= axp_init_irq,
 	.timer		= &axp_timer,
 	.init_machine	= axp_rd_nas_init,
+	.restart	= arch_reset,
 #ifdef CONFIG_FB_DOVE
 	/* reserve memory for LCD */
 	.fixup		= axp_tag_fixup_mem32,
@@ -2036,6 +2039,7 @@ MACHINE_START(ARMADA_XP_RDSRV, "Marvell Armada XP Server Board")
   	.init_irq	= axp_init_irq,
   	.timer		= &axp_timer,
   	.init_machine	= axp_rdsrv_init,
+	.restart	= arch_reset,
   MACHINE_END
 
 /*****************************************************************************
@@ -2106,4 +2110,5 @@ MACHINE_START(ARMADA_XP_FPGA, "Marvell Armada XP FPGA Board")
 	.init_irq	= axp_init_irq,
 	.timer		= &axp_timer,
 	.init_machine	= axp_fpga_init,
+	.restart	= arch_reset,
 MACHINE_END
diff --git a/arch/arm/mach-armadaxp/include/mach/system.h b/arch/arm/mach-armadaxp/include/mach/system.h
index a359b34..167201b 100644
--- a/arch/arm/mach-armadaxp/include/mach/system.h
+++ b/arch/arm/mach-armadaxp/include/mach/system.h
@@ -29,7 +29,6 @@ static inline void arch_idle(void)
 
 static inline void arch_reset(char mode, const char *cmd)
 {
-	printk("Reseting...\n");
 	mvBoardReset();
 	while (1);/* This should never be reached */
 }
-- 
1.7.0

